perceval/utils/logging/loggers.py

```diff
@@ -123,6 +126,13 @@ def _configure_logger(self):
                     exq_log.level.__members__[level],
                     exq_log.channel.__members__[channel])
 
+    def apply_config(self, config: LoggerConfig):
```

Comment:
To avoid the inability to change the logger type, from inside a given logger, you can define an `apply_logger_config()` function outside of the logger classes. It would remove both user warnings.